gobblin-yarn/src/test/java/gobblin/yarn/GobblinHelixTaskTest.java

```diff
@@ -0,0 +1,177 @@
+    JobState jobState = new JobState();
+    jobState.setJobName(TestHelper.TEST_JOB_NAME);
+    jobState.setJobId(TestHelper.TEST_JOB_ID);
+    SerializationUtils.serializeState(this.localFs, jobStateFilePath, jobState);
+
+    // Prepare the WorkUnit
+    WorkUnit workUnit = WorkUnit.createEmpty();
+    prepareWorkUnit(workUnit);
+
+    // Prepare the source Json file
+    File sourceJsonFile = new File(this.appWorkDir.toString(), TestHelper.TEST_JOB_NAME + ".json");
+    TestHelper.createSourceJsonFile(sourceJsonFile);
+    workUnit.setProp(SimpleJsonSource.SOURCE_FILE_KEY, sourceJsonFile.getAbsolutePath());
+
+    // Serialize the WorkUnit into a file
+    Path workUnitFilePath = new Path(this.appWorkDir, TestHelper.TEST_JOB_NAME + ".wu");
+    SerializationUtils.serializeState(this.localFs, workUnitFilePath, workUnit);
+    Assert.assertTrue(this.localFs.exists(workUnitFilePath));
+
+    // Prepare the GobblinHelixTask
+    Map<String, String> taskConfigMap = Maps.newHashMap();
+    taskConfigMap.put(GobblinYarnConfigurationKeys.WORK_UNIT_FILE_PATH, workUnitFilePath.toString());
+    taskConfigMap.put(ConfigurationKeys.JOB_ID_KEY, TestHelper.TEST_JOB_ID);
+
+    TaskConfig taskConfig = new TaskConfig("", taskConfigMap, true);
+    TaskCallbackContext taskCallbackContext = Mockito.mock(TaskCallbackContext.class);
+    Mockito.when(taskCallbackContext.getTaskConfig()).thenReturn(taskConfig);
+    Mockito.when(taskCallbackContext.getManager()).thenReturn(this.helixManager);
+
+    GobblinHelixTaskFactory gobblinHelixTaskFactory =
+        new GobblinHelixTaskFactory(this.taskExecutor, this.taskStateTracker, this.localFs, this.appWorkDir);
+    this.gobblinHelixTask = (GobblinHelixTask) gobblinHelixTaskFactory.createNewTask(taskCallbackContext);
+  }
+
+  @Test(dependsOnMethods = "testPrepareTask")
+  public void testRun() throws IOException {
+    TaskResult taskResult = this.gobblinHelixTask.run();
+    System.out.println(taskResult.getInfo());
+    Assert.assertEquals(taskResult.getStatus(), TaskResult.Status.COMPLETED);
+
+    File outputAvroFile = new File(this.taskOutputDir.toString(),
+        TestHelper.REL_WRITER_FILE_PATH + File.separator + TestHelper.WRITER_FILE_NAME);
+    Assert.assertTrue(outputAvroFile.exists());
+
+    Schema schema = new Schema.Parser().parse(TestHelper.SOURCE_SCHEMA);
+
+    try (DataFileReader<GenericRecord> reader =
+        new DataFileReader<>(outputAvroFile, new GenericDatumReader<GenericRecord>(schema))) {
+      Iterator<GenericRecord> iterator = reader.iterator();
+
+      GenericRecord record = iterator.next();
+      Assert.assertEquals(record.get("name").toString(), "Alyssa");
+
+      record = iterator.next();
+      Assert.assertEquals(record.get("name").toString(), "Ben");
+
+      record = iterator.next();
+      Assert.assertEquals(record.get("name").toString(), "Charlie");
+
+      Assert.assertFalse(iterator.hasNext());
```

Comment:
Move above code into a method as it seems to be a duplicate of code in `GobblinHelixJobLauncherTest`